### PR TITLE
koordlet: support change pod CPUQOS by annotations

### PR DESCRIPTION
**Pull request by j4ckstraw**
### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

**Comment by zwzhang0107** [comment]
seems we need some UT for this

File: apis/slo/v1alpha1/pod.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ import (
 const (
 	AnnotationPodCPUBurst = apiext.DomainPrefix + "cpuBurst"
 
+	AnnotationPodCPUQoS = apiext.DomainPrefix + "CPUQOS"
```
Comment (zwzhang0107) [inline apis/slo/v1alpha1/pod.go:30]: apiext.DomainPrefix + "cpuQOS"

File: pkg/koordlet/runtimehooks/hooks/groupidentity/interceptor.go [hunk 2]
```diff
@@ -17,14 +17,39 @@ limitations under the License.
...
 	"k8s.io/klog/v2"
 	"k8s.io/utils/pointer"
 
 	ext "github.com/koordinator-sh/koordinator/apis/extension"
+	slov1alpha1 "github.com/koordinator-sh/koordinator/apis/slo/v1alpha1"
 	"github.com/koordinator-sh/koordinator/pkg/koordlet/runtimehooks/protocol"
 	"github.com/koordinator-sh/koordinator/pkg/koordlet/util"
 )
 
+func getPodCPUQoSConfig(annotations map[string]string) (*slov1alpha1.CPUQOSCfg, error) {
```
Comment (zwzhang0107) [inline pkg/koordlet/runtimehooks/hooks/groupidentity/interceptor.go:32]: better to mv this func to api/slo/v1aplha1/pod.go

File: pkg/koordlet/runtimehooks/hooks/groupidentity/interceptor.go [hunk 3]
```diff
@@ -35,6 +60,27 @@ func (b *bvtPlugin) SetPodBvtValue(p protocol.HooksProtocol) error {
 	podQOS := ext.GetQoSClassByAttrs(req.Labels, req.Annotations)
 	podKubeQOS := util.GetKubeQoSByCgroupParent(req.CgroupParent)
 	podBvt := r.getPodBvtValue(podQOS, podKubeQOS)
+
+	// pod annotatations take precedence to the default CPUQoS
+	cfg, err := getPodCPUQoSConfig(req.Annotations)
```
Comment (zwzhang0107) [inline pkg/koordlet/runtimehooks/hooks/groupidentity/interceptor.go:65]: better to use DefaultCPUQOS() as default and merge with annotation, besides, cfg.Enable == nil means disabled by default

File: pkg/koordlet/runtimehooks/hooks/groupidentity/interceptor.go (hunk 2 above)
Comment (j4ckstraw) [reply pkg/koordlet/runtimehooks/hooks/groupidentity/interceptor.go:32]: this function is different to exist functions like `GetPodMemoryQoSConfig`, which receive parameter from annotation, so I define it in local file. If we move it to `apis/slo/v1aplha1/pod.go`, how about named `GetPodCPUQoSConfigFromAnnotations`?

Comment (j4ckstraw) [reply pkg/koordlet/runtimehooks/hooks/groupidentity/interceptor.go:32]: or `GetPodCPUQoSConfigByAttrs`?

**Pull request merged**